compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -407,6 +407,30 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             sym::transmute => {
                 self.copy_op_transmute(args[0], dest)?;
             }
+            sym::assert_inhabited | sym::assert_zero_valid | sym::assert_uninit_valid => {
```

Comment:
@oli-obk how will such aborting intrinsics interact with function memoization?

Reply:
In particular with https://github.com/rust-lang/rust/issues/75390 I am wondering if we'll get decent error messages here.

Reply:
that works fine, it will cause the const eval queries to return an error, similar to how the memoization of `size_of` can error if the type is not representable on the platform